added PrintEventReceiver for easy debug logging

bradleypeabody opened the pull request:
I didn't see any way to easily enable debugging logging.  There is a NullEventReceiver which throws everything away, but I don't see anything that provides simple logging.

I made a PrintEventReceiver which writes to anything with a Print() method, including a log.Logger.  So now you can do:

```
	// write debug to os.Stderr
	sess := conn.NewSession(dbr.NewPrintEventReceiver(nil))
```

or, for more control:

```
	sess := conn.NewSession(
		dbr.NewPrintEventReceiver(
			log.New(os.Stdout, "SQL DEBUG: ", log.LstdFlags),
		),
	)
```

What do you think?  Any reason not to merge this in?

taylorchu commented:
I was considering [zap](https://github.com/uber-go/zap) or logrus at some point, and put it in a sub-package. Logging requirement is often unique to the problem that you solve.

bradleypeabody commented:
@taylorchu That's a good point - definitely there can be a lot of variation when you need detailed logging.

From my experience, there is also a common case of simply "show me the SQL statements, I need to see what is happening".  This is not really easy to do without something like PrintEventReceiver.

Also, logrus implements the Printer interface I added, so you get that support for free:

```
sess := conn.NewSession(dbr.NewPrintEventReceiver(logrus.New()))
```

And I added a PrinterFunc, so you can easily adapt other log packages, including zap:

```
sugaredLogger := zap.NewExample().Sugar()
sess := conn.NewSession(dbr.NewPrintEventReceiver(
	dbr.PrinterFunc(func(v ...interface{}) { sugaredLogger.Debug(v...) })
))
```

Better? :)

(I realize you are concerned about the more complicated cases, but I think it is also good to make the simple cases simple, i.e. the "just show me all the SQL in the log" case.)
